### PR TITLE
footer links from contentful

### PR DESCRIPTION
**Pull request by lbiedinger**

### CODE REVIEW

File: components/PageFooter.vue [hunk 1]
```diff
@@ -1,9 +1,68 @@
 <template>
-  <footer>{{ $t('footerText') }}</footer>
```
Comment (renekok) [inline components/PageFooter.vue]: remove footerText from locale files

File: components/PageFooter.vue [hunk 2]
```diff
@@ -1,9 +1,68 @@
...
+          :key="footerLink.url"
+        >
+          <b-link
+            v-if="footerLink.url"
+            :href="footerLink.url"
+            class="footer-link"
+          >
+            {{ footerLink.text }}
+          </b-link>
+          <p v-else>
```
[the remark this answers is not in the document]
Comment (lbiedinger) [reply components/PageFooter.vue]: it serves 2 functions as of now:
* a hacky way to output any errors, while no other error handling is in place
* future proofing for a potential case where footer elements are to be added that aren't links. (currently the content model wouldn't allow this anyway)

File: store/footer.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import contentfulClient from '../plugins/contentful.js';
+
+export const state = () => ({
+  links: null
```
Comment (rwd) [inline store/footer.js]: Initialise this to `[]` instead of `null`

File: store/footer.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+export const actions = {
+  async init ({ commit }) {
+    let data = await contentfulClient.getEntries({
+      'content_type': 'linkGroup',
+      'fields.identifier': 'footer',
+      'limit': 1
+    })
+      .then((response) => {
+        if (response.total == 0) {
+          return [];
```
Comment (rwd) [inline store/footer.js]: This can be omitted if the default state value is made to be `[]`

File: store/index.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export const state = () => ({
```
Comment (rwd) [inline store/index.js]: This seems redundant and can be removed.

File: tests/unit/components/PageFooter.spec.js [hunk 6]
```diff
@@ -1,17 +1,39 @@
...
-  it('it exists', () => {
+  it('it contains elements for each link', async () => {
     const wrapper = factory();
-
-    const footer = wrapper.find('footer');
-    footer.isVisible().should.equal(true);
+    await sleep(20);
+    const renderedList = wrapper.findAll('footer ul li');
+    renderedList.length.should.eq(2);
+    //dummyLinkGroup.items[0].links[0].fields.url
```
Comment (rwd) [inline tests/unit/components/PageFooter.spec.js]: Remove this

File: tests/unit/components/PageFooter.spec.js [hunk 7]
```diff
@@ -1,17 +1,39 @@
...
+};
+
 describe('components/search/PageFooter', () => {
-  it('it exists', () => {
+  it('it contains elements for each link', async () => {
     const wrapper = factory();
-
-    const footer = wrapper.find('footer');
-    footer.isVisible().should.equal(true);
+    await sleep(20);
```
Comment (rwd) [inline tests/unit/components/PageFooter.spec.js]: Why is this needed? If I remove it, the test still passes.

File: store/footer.js (hunk 4 above)
Comment (lbiedinger) [reply store/footer.js]: it still needs to check for response.total == 0, otherwise it'll throw an error when someone deletes/unpublishes the 'footer' linkGroup.

File: tests/unit/components/PageFooter.spec.js (hunk 7 above)
Comment (lbiedinger) [reply tests/unit/components/PageFooter.spec.js]: Not needed any more, it was needed when the links were only retrieved on the client as that happened only after the component was rendered.

Have removed it.

File: nuxt.config.js [hunk 8]
```diff
@@ -43,7 +43,7 @@ module.exports = {
   /*
   ** Plugins to load before mounting the App
   */
-  plugins: ['~/plugins/vue-filters'],
+  plugins: ['~/plugins/vue-filters', { src: './plugins/global.js' }],
```
Comment (rwd) [inline nuxt.config.js]: Let's keep this consistent and concise, with:
```js
plugins: ['~/plugins/vue-filters', '~/plugins/global']
```

File: components/generic/Hyperlink.vue [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    :class="linkClass"
+  >
+    <slot />
+  </b-link>
+</template>
+
+<script>
+  export default {
+    props: {
+      link: {
```
Comment (rwd) [inline components/generic/Hyperlink.vue]: Could we name this something more descriptive, perhaps "destination"?

File: plugins/global.js [hunk 10]
```diff
@@ -0,0 +1,4 @@
+import Vue from 'vue';
+import Hyperlink from '../components/generic/Hyperlink.vue';
```
Comment (rwd) [inline plugins/global.js]: > Component names should always be multi-word

https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential
